apps/web/src/server/lib/msal.js

```diff
@@ -1,6 +1,29 @@
 import msal, { LogLevel } from '@azure/msal-node';
 import config from '@pins/applications.web/environment/config.js';
 import pino from './logger.js';
+import * as redis from './redis.js';
+
+/**
+ * @param {*} cacheContext
+ * */
+const beforeCacheAccess = async (cacheContext) => {
+	const cacheData = await redis.get('msal');
+	cacheContext.tokenCache.deserialize(cacheData);
+};
+
+/**
+ * @param {*} cacheContext
+ * */
+const afterCacheAccess = async (cacheContext) => {
+	if (cacheContext.cacheHasChanged) {
+		await redis.set('msal', cacheContext.tokenCache.serialize());
+	}
+};
+
+const cachePlugin = {
+	beforeCacheAccess,
+	afterCacheAccess
+};
```

Comment:
We might want to move this to a shared package and make it a class, that takes a redis client. That way appeals can use it too.

Not required though if we want to check it works first.